code/game/striketeams/striketeam_datums.dm

```diff
@@ -0,0 +1,185 @@
+
+	searching = TRUE
+
+	var/icon/team_logo = icon('icons/mob/mob.dmi', logo)
+	for(var/mob/dead/observer/O in dead_mob_list)
+		if(jobban_isbanned(O, "Strike Team") || O.client.is_afk())
+			continue
+
+		to_chat(O, "[bicon(team_logo)]<span class='recruit'>[faction_name] needs YOU to become part of its upcoming [striketeam_name]. (<a href='?src=\ref[src];signup=\ref[O]'>Apply now!</a>)</span>[bicon(team_logo)]")
+
+	spawn(600)
+		searching = FALSE
+
+		for(var/mob/dead/observer/O in dead_mob_list)
+			if(jobban_isbanned(O, "Strike Team") || O.client.is_afk())
+				continue
+			to_chat(O, "[bicon(team_logo)]<span class='recruit'>Applications for [faction_name]'s [striketeam_name] are now closed.</span>[bicon(team_logo)]")
+
+		if(!applicants || applicants.len <= 0)
+			if(user)
+				to_chat(user, "<span class='warning'>[faction_name] received no applications for the [striketeam_name].</span>")
+			failure()
+			qdel(src)
+			return
+
+		var/list/commando_spawns = list_commando_spawns()
+		var/commando_count = min(applicants.len,team_size)
+		var/leader = FALSE
+
+		if(min_size_for_leader >= 0 && applicants.len >= min_size_for_leader)
+			leader = TRUE
+
+		for (var/i = commando_count, i > 0, i--)
+			if (commando_spawns.len <= 0)
+				commando_spawns = list_commando_spawns()
+				//The point here is to try not having commandos spawn atop each others.
+
+			if(applicants.len <= 0)
+				break
+
+			var/mob/applicant = null
+			for(var/mob/M in player_list)
+				if(M.key == pick(applicants))
+					applicant = M
+			applicants -= applicant.key
+
+			if(!istype(applicant, /mob/dead/observer))
+				//Making sure we don't recruit people who got back into the game since they applied
+				continue
+
+			var/obj/effect/landmark/L = pick(commando_spawns)
+			commando_spawns -= L
+
+			spawn()//not waiting for players to customize their characters to move on
+				var/mob/living/carbon/human/new_commando = create_commando(L, leader, applicant.key)
+				team_composition |= new_commando
+				new_commando.key = applicant.key
+
+				if (leader)
+					leader_key = new_commando.key
+					leader = FALSE
+
+				new_commando.mind.store_memory("<B>Mission:</B> <span class='warning'>[mission].</span>")
+
+				greet_commando(new_commando)
+		extras()
+
+/datum/striketeam/proc/create_commando(var/obj/spawn_location,var/leader_selected=0,var/mob_key = "")
+	var/mob/living/carbon/human/new_commando = new(spawn_location.loc)
+	return new_commando
+
+/datum/striketeam/proc/greet_commando(var/mob/living/carbon/human/H)
+	to_chat(H, "<span class='notice'>You are a [striketeam_name] commando, in the service of [faction_name].</span>")
+	to_chat(H, "<span class='notice'>Your current mission is: <span class='danger'>[mission]</span></span>")
+
+/datum/striketeam/Topic(var/href, var/list/href_list)
+	if(href_list["signup"])
+		var/mob/dead/observer/O = locate(href_list["signup"])
```

Comment:
You shouldn't use the href to locate the observer here, go by `usr` instead.

Right now this would allow you to make anybody sign up if you knew the ref of your fellow ghosts.